pkg/util/cluster/aad.go

```diff
@@ -61,6 +61,23 @@ func (c *Cluster) createServicePrincipal(ctx context.Context, appID string) (str
 }
 
 func (c *Cluster) deleteApplication(ctx context.Context, appID string) error {
+
+	// Clean up the role assignments for the service principal first
+	spObjID, err := c.getServicePrincipal(ctx, appID)
+	if err != nil {
+		return err
+	}
+
+	roleAssignments, err := c.roleassignments.List(ctx, fmt.Sprintf("principalId eq '%s'", spObjID))
+	if err != nil {
+		return err
+	}
+
+	c.log.Info("deleting role assignments")
+	for _, roleAssignment := range roleAssignments {
+		c.roleassignments.Delete(ctx, *roleAssignment.Scope, *roleAssignment.Name)
```

Comment:
should be a DeleteAndWait.  Check and handle the error returned (always).

Reply:
the roleAssignmentsClient doesn't reuturn a Future.  This mimics the behavior of the application client.

Reply:
oh yeah :-)